### PR TITLE
Inventory item details ticket 21

### PR DESCRIPTION
**Pull request by zahrafalak**
**Ticket Description:**
The user must be able to view the details for a specific inventory item when an item is clicked from the list view. The user must have the ability to navigate to the edit item functionality from this view. They should also be able to navigate back to the inventory list.

Create the UI and functionality for displaying the details of a specific inventory item.

Ensure this component works at and between all breakpoints and is fully responsive without any elements overlapping.

**Comment(s):**
The page navigates to /inventory/edit when the edit button is clicked, but the edit page does not exist yet. The page also navigates back to the inventory list component when the back icon is clicked. 

Screenshot 2024-03-26 at 10 32 55 PM

Screenshot 2024-03-26 at 10 33 07 PM

### CODE REVIEW

**Comment by ryn-ngn** [comment]
Hi Falak, in general the code works. However, there are two things to consider:

1. After the refactoring come into effective, things will need to be fixed here and there
2. Rem is almost always prefered instead of px in general but for the scope of this project, i guess it should be no problem.

**Comment by zahrafalak** [comment]
Yeah I just used px (a lot more than I have ever done) to follow the Figma thing as close as possible. Didn't seems to make too much difference in the scope of the project :)

**Review by Elizabeth-Sorescu — APPROVED** [review]
Good job, Falak!

File: src/components/Inventory/InventoryDetails.js [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+          `http://localhost:8080/api/inventories/${id}`
+        );
+        setInventoryDetails(resp.data);
+        setDetailsLoaded(true);
+      } catch (error) {
+        console.error("Error fetching inventory data:", error);
+      }
+    };
+    fetchInventoryDetails();
+  });
```
Comment (rjtbansal) [inline src/components/Inventory/InventoryDetails.js:35]: You are missing dependency array here. And thats why the network call to this page is infinite and its almost crashing my page.
Always have network tab open to ensure these things are not missed.

 ```
fetchInventoryDetails();
  }, []);
```

Comment (rjtbansal) [reply src/components/Inventory/InventoryDetails.js:35]: cc: @zahrafalak

**Review by rjtbansal — COMMENTED** [review]
UI looks good on all screens!

But the missing dependency array from the useEffect is considerably slowing down my browser

**Pull request merged**